backend/battle/tests/test_api.py

```diff
@@ -0,0 +1,348 @@
+                'creator': self.user_1.email,
+            },
+        )
+
+
+class CreateTeamEndpointTest(APITestCaseUtils):
+    def setUp(self):
+        super().setUp()
+        self.battle = baker.make('battle.Battle', creator=self.user_1)
+        names = ['pikachu', 'bulbasaur', 'pidgeot', 'wrong_name']
+        for pokemon_name in names:
+            baker.make("pokemon.Pokemon", name=pokemon_name, attack=60, defense=15, hp=10)
+
+    @patch('battle.views.run_battle_and_send_result_email.delay')
+    def test_if_task_run_battle_is_called(self, task_mock):
+        battle = Battle.objects.create(creator=self.user_1, opponent=self.opponent)
+
+        pokemons_data_opponnent = {
+            "battle": battle.id,
+            "trainer": self.opponent.id,
+            "pokemon_1": 'pikachu',
+            "pokemon_2": 'bulbasaur',
+            "pokemon_3": 'pidgeot',
+            "position_pkn_1": 1,
+            "position_pkn_2": 2,
+            "position_pkn_3": 3,
+        }
+        self.auth_client.post(
+            reverse("team-create"), pokemons_data_opponnent, follow=True)
+        self.assertTrue(Team.objects.get(battle=battle, trainer=self.opponent))
+
+        pokemons_data = {
+            "battle": battle.id,
+            "trainer": self.user_1.id,
+            "pokemon_1": 'pikachu',
+            "pokemon_2": 'bulbasaur',
+            "pokemon_3": 'pidgeot',
+            "position_pkn_1": 1,
+            "position_pkn_2": 2,
+            "position_pkn_3": 3,
+        }
+        self.auth_client.post(
+            reverse("team-create"), pokemons_data, follow=True)
+
+        task_mock.assert_called_with(battle.id)
+
+        self.assertTrue(Team.objects.get(battle=battle, trainer=self.user_1))
+
+    @patch('battle.battles.email.send_templated_mail')
+    def test_if_task_send_email(self, email_mock):
+        battle = Battle.objects.create(creator=self.user_1, opponent=self.opponent)
+
+        pokemons_data_opponnent = {
+            "battle": battle.id,
+            "trainer": self.opponent.id,
+            "pokemon_1": 'pikachu',
+            "pokemon_2": 'bulbasaur',
+            "pokemon_3": 'pidgeot',
+            "position_pkn_1": 1,
+            "position_pkn_2": 2,
+            "position_pkn_3": 3,
+        }
+        self.auth_client.post(
+            reverse("team-create"), pokemons_data_opponnent, follow=True)
+        self.assertTrue(Team.objects.get(battle=battle, trainer=self.opponent))
+
+        pokemons_data = {
+            "battle": battle.id,
+            "trainer": self.user_1.id,
+            "pokemon_1": 'pikachu',
+            "pokemon_2": 'bulbasaur',
+            "pokemon_3": 'pidgeot',
+            "position_pkn_1": 1,
+            "position_pkn_2": 2,
+            "position_pkn_3": 3,
+        }
+        self.auth_client.post(
+            reverse("team-create"), pokemons_data, follow=True)
+
+        battle_updated = Battle.objects.get(creator=self.user_1, opponent=self.opponent)
```

Comment:
If you want to get an updated version of the battle you can use: `battle.refresh_from_db()`

Reply:
great!